validator/project_validator_test.go

```diff
@@ -3284,6 +3302,28 @@ func TestDependencyMustRun(t *testing.T) {
 			},
 			expectDependencyFound: false,
 		},
+		"DependencySkipsGitTagsIfNotAllowedForGitTags": {
+			source: model.TVPair{TaskName: "A", Variant: "ubuntu"},
+			target: model.TVPair{TaskName: "B", Variant: "ubuntu"},
+			depReqs: dependencyRequirements{
+				lastDepNeedsSuccess: true,
+				requireOnPatches:    false,
+				requireOnNonPatches: false,
+				requireOnGitTag:     true,
+			},
+			tvToTaskUnit: map[model.TVPair]model.BuildVariantTaskUnit{
+				{TaskName: "A", Variant: "ubuntu"}: {
+					Patchable: falsePtr,
+					DependsOn: []model.TaskUnitDependency{
+						{Name: "B", Variant: "ubuntu"},
+					},
+				},
+				{TaskName: "B", Variant: "ubuntu"}: {
+					AllowForGitTag: falsePtr,
+				},
+			},
+			expectDependencyFound: false,
```

Comment:
Can we test for the case of `requireOnGitTag` is true and `AllowForGitTag` is also true?

Reply:
Done